scipy/linalg/matfuncs.py

```diff
@@ -668,3 +675,329 @@ def rounded_sign(x):
+    Parameters
+    ----------
+    A : (N, N) array_like
+        Matrix to evaluate.
+    tol : float
+        Tolerance for the zero value.
+    
+    Returns
+    -------
+    X : boolean
+        True if a matrix A is positive semidefinite.
+
+    Examples
+    --------
+    >>> from scipy.linalg import is_matrix_positive_semidefinite
+    >>> m = np.array([[1,-1],[-1,1]])
+    >>> print(is_matrix_positive_semidefinite(m))
+    True
+    """
+    A = _asarray_square(A)
+    if not is_matrix_hermitian(A):
+        raise ValueError('expected symmetric or hermitian matrix')
+    eiVal, eiVecR = eig(A)
+    eiVal[np.absolute(eiVal) < tol] = 0
+    return (eiVal >= 0).all()
+
+
+def is_matrix_negative_definite(A, tol=1e-8):
+    """
+    Returns True if a matrix is negative definite
+    Parameters
+    ----------
+    A : (N, N) array_like
+        Matrix to evaluate.
+    tol : float
+        Tolerance for the zero value.
+    
+    Returns
+    -------
+    X : boolean
+        True if a matrix A is negative definite.
+
+    Examples
+    --------
+    >>> from scipy.linalg import is_matrix_negative_definite
+    >>> m = np.array([[-5,-1],[-1,-3]])
+    >>> print(is_matrix_negative_definite(m))
+    True
+    """
+    A = _asarray_square(A)
+    if not is_matrix_hermitian(A):
+        raise ValueError('expected symmetric or hermitian matrix')
+    eiVal, eiVecR = eig(A)
+    eiVal[np.absolute(eiVal) < tol] = 0
+    return (eiVal < 0).all()
+
+
+def is_matrix_negative_semidefinite(A, tol=1e-8):
+    """
+    Returns True if a matrix is negative semidefinite
+    Parameters
+    ----------
+    A : (N, N) array_like
+        Matrix to evaluate.
+    tol : float
+        Tolerance for the zero value.
+    
+    Returns
+    -------
+    X : boolean
+        True if a matrix A is negative semidefinite.
+
+    Examples
+    --------
+    >>> from scipy.linalg import is_matrix_negative_semidefinite
+    >>> m = np.array([[-1,1],[1,-1]])
+    >>> print(is_matrix_negative_semidefinite(m))
+    True
+    """
+    A = _asarray_square(A)
+    if not is_matrix_hermitian(A):
+        raise ValueError('expected symmetric or hermitian matrix')
+    eiVal, eiVecR = eig(A)
+    eiVal[np.absolute(eiVal) < tol] = 0
+    return (eiVal <= 0).all()
+
+
+def is_matrix_indefinite(A, tol=1e-8):
+    """
+    Returns True if a matrix is indefinite
+    Parameters
+    ----------
+    A : (N, N) array_like
+        Matrix to evaluate.
+    tol : float
+        Tolerance for the zero value.
+    
+    Returns
+    -------
+    X : boolean
+        True if a matrix A is indefinite.
+
+    Examples
+    --------
+    >>> from scipy.linalg import is_matrix_indefinite
+    >>> m = np.array([[5,1],[1,0]])
+    >>> print(is_matrix_indefinite(m))
+    True
+    """
+    A = _asarray_square(A)
+    if not is_matrix_hermitian(A):
+        raise ValueError('expected symmetric or hermitian matrix')
+    eiVal, eiVecR = eig(A)
+    eiVal[np.absolute(eiVal) < tol] = 0
```

Comment:
this check would not be used since it is a one liner if we already have the eigenvalues and also fails for zero matrix.

Reply:
It should be fine now.